### PR TITLE
feat(configparser): update to 3.11

### PR DESCRIPTION
**Pull request by carlosmiei**
- Lib: https://github.com/python/cpython/blob/3.11/Lib/configparser.py
- Test: https://github.com/python/cpython/blob/3.11/Lib/test/test_configparser.py

Test

```
RUSTPYTHONPATH=Lib cargo run --release Lib/test/test_configparser.py
    Updating git repository `https://github.com/youknowone/unicode_names2.git`
    Finished release [optimized] target(s) in 0.92s
     Running `target/release/rustpython Lib/test/test_configparser.py`
...
Ran 343 tests in 1.312s

OK (skipped=5)
```

### CODE REVIEW

File: Lib/test/test_configparser.py [hunk 1]
```diff
@@ -521,7 +528,6 @@ def test_default_case_sensitivity(self):
             cf.get(self.default_section, "Foo"), "Bar",
             "could not locate option, expecting case-insensitive defaults")
 
-    @unittest.skipIf(os.name == "nt", "TODO: RUSTPYTHON, universal newlines")
```
Comment (DimitrisJim) [inline Lib/test/test_configparser.py:524]: probably need to add this again, it skipped the tests for windows (which are now failing)

**Comment by fanninpm** [comment]
When I update a library from a newer version of CPython, I usually do this:

1. Copy/paste the library straight from CPython unless it has any RustPython-related annotations, and put that into the first commit. (Your first commit is fine as is, but personally, I would title it `Update configparser.py from CPython 3.11` in accordance with [this guide](https://commit.style).)
2. Copy/paste the test straight from CPython, _without touching the RustPython-related annotations_, and put that into a second commit, titled something like `Update test_configparser.py from CPython 3.11`. (You can follow [the guide in the wiki](https://github.com/RustPython/RustPython/wiki/How-to-update-test-files), but I use Meld for this.)
3. Run the new test. Add/remove any necessary RustPython-related annotations, and put that into one or more commits, titled something like `Mark failing/erroring tests` or `Unmark passing tests` or `Skip panicking test`.

**Comment by youknowone** [comment]
#4597 will make it simpler

**Comment by carlosmiei** [comment]
> When I update a library from a newer version of CPython, I usually do this:
> 
> 1. Copy/paste the library straight from CPython unless it has any RustPython-related annotations, and put that into the first commit. (Your first commit is fine as is, but personally, I would title it `Update configparser.py from CPython 3.11` in accordance with [this guide](https://commit.style).)
> 2. Copy/paste the test straight from CPython, _without touching the RustPython-related annotations_, and put that into a second commit, titled something like `Update test_configparser.py from CPython 3.11`. (You can follow [the guide in the wiki](https://github.com/RustPython/RustPython/wiki/How-to-update-test-files), but I use Meld for this.)
> 3. Run the new test. Add/remove any necessary RustPython-related annotations, and put that into one or more commits, titled something like `Mark failing/erroring tests` or `Unmark passing tests` or `Skip panicking test`.

@fanninpm thanks for your suggestions, I was trying to follow `conventional commits` convention but will stick with your suggestion next time.

> Copy/paste the test straight from CPython, without touching the RustPython-related annotations

Sorry I'm a bit confused here, let's say the test has some annotations, if I'm copying it straight from CPython how would I not touch those?

File: Lib/test/test_configparser.py (hunk 1 above)
Comment (carlosmiei) [reply Lib/test/test_configparser.py]: @DimitrisJim restored thanks

**Pull request merged**